### ISSUE TITLE
[Feature] Detect lost WAR (write-after-read) dependencies in task scheduling and fix via inout rewrite or explicit error

### ISSUE THREAD

**Issue by YunjiQin** [hw-native-sys/pypto#2058]
### Summary

The runtime task scheduler drops **WAR (write-after-read) anti-dependencies** for loop-carried buffers, which silently produces data races and precision/accuracy errors. PyPTO currently has no mechanism to detect or guard against this. We should add **WAR analysis** to the task-dependency derivation path so that, when a WAR hazard is detected, PyPTO either:

1. rewrites the reading task's `Input` argument to `InOut` so the runtime tracks it as a new writer version and emits the missing edge, **or**
2. raises a clear compile-time error instructing the user to construct the WAR dependency explicitly via the manual-dependency mechanism (`pl.submit(..., deps=[...])`).

Related runtime bug: hw-native-sys/simpler#1306. Related PyPTO RFC on automatic dependency derivation: #1368.

### Motivation / Use Case

The runtime dependency generator emits RAW and WAW edges but **omits WAR edges** in loop-carried buffer scenarios. Concretely, for a buffer allocated once and carried across loop iterations:

- Iteration N: `writer(N)` modifies the buffer (`inout`), then `reader(N)` reads it (pure `input`).
- Iteration N+1: `writer(N+1)` modifies the buffer again.

The runtime generates:
- RAW: `writer(N) -> reader(N)` (same iteration), and
- WAW: `writer(N) -> writer(N+1)` (loop-carried),

but **not** the required WAR edge `reader(N) -> writer(N+1)`. Because the reader produces no new version of the buffer, the loop carries the *writer's* output forward, so `writer(N+1)` depends only on `writer(N)` — never on `reader(N)`. The scheduler is then free to run `writer(N+1)` concurrently with `reader(N)`, overwriting the buffer while it is still being read → data race.

**Impact (from the runtime issue):** in a minimal reproducer ~35% of one band's output contained another band's data; in production (Qwen3-14B prefill) ~10% of logits were corrupted due to shared gate/accumulator buffers racing across MLP bands.

**Why this needs a PyPTO-side fix:** the corruption is silent — no crash, just wrong numbers — and is extremely hard to diagnose because the generated orchestration looks correct (all directions/annotations are right; only the dependency edge is missing). PyPTO owns argument-direction assignment (`DeriveCallDirections`) and compiler dependency derivation (`AutoDeriveTaskDependencies`), so it is the right layer to detect the hazard and either encode the missing constraint or fail loudly. Today PyPTO has neither WAR analysis nor a diagnostic for this case, so users hit silent precision errors with no guidance.

### Proposed API / Behavior

Add WAR-hazard analysis to the dependency-derivation path (candidate home: `AutoDeriveTaskDependencies`, pass 35, which already builds a storage-location/alias map for tensor Vars and tracks loop-carried roots). For each storage root that is:

- read as pure `Input` by a task in iteration N, and
- written (`inout`/output) by a task in iteration N+1 through the loop carry,

detect the missing `reader(N) -> writer(N+1)` WAR edge and resolve it via one of:

```text
Option A (automatic):  rewrite the reader's Input arg on that slot to InOut
                       so the runtime versions the buffer and derives the edge.
Option B (diagnostic): raise a clear compile-time error naming the buffer and
                       the two tasks, instructing the user to add the WAR edge
                       explicitly:

    # reader(N) must complete before writer(N+1) overwrites `buf`
    _, tid_read  = pl.submit(self.reader, buf, ...)
    _, tid_write = pl.submit(self.writer, buf, ..., deps=[tid_read])
```

Behavior should be scoped to AUTO runtime scopes (MANUAL scopes already require explicit `deps=[...]` and must keep their scheduling contract untouched). A conservative default of Option B (error out) is acceptable as a first step so correctness is never silently violated; Option A (automatic `Input -> InOut` rewrite) can follow as an optimization once the analysis is proven sound.

### Alternatives Considered

- **Fix only in the runtime scheduler** (hw-native-sys/simpler#1306): necessary but not sufficient — PyPTO should still surface/guard the hazard so users on current runtime versions are not silently miscompiled, and so the constraint is expressed in the IR/orchestration.
- **Always require manual `deps=[...]`**: too burdensome and easy to forget; the whole point of AUTO scopes is automatic dependency derivation. A detect-and-guard/auto-fix approach preserves that ergonomics.
- **Do nothing (status quo)**: unacceptable — produces silent, hard-to-debug precision errors in real models.

### Additional Context

- Runtime issue: hw-native-sys/simpler#1306 (WAR dependency loss in the runtime task scheduler).
- Related PyPTO work: #1368 (RFC: automatic manual-scope dependency derivation), #1744, #1939.
- Relevant PyPTO code: `AutoDeriveTaskDependencies` (pass 35, `docs/en/dev/passes/35-auto_derive_task_dependencies.md`) and `DeriveCallDirections` (pass 34); manual mechanism `pl.submit(..., deps=[...])`.
